Java/core/src/main/java/com/amazon/randomcutforest/executor/PointStoreCoordinator.java

```diff
@@ -21,31 +21,34 @@
 
 import com.amazon.randomcutforest.store.IPointStore;
 
-public class PointStoreCoordinator<Q> extends AbstractUpdateCoordinator<Integer, Q> {
+public class PointStoreCoordinator<Point> extends AbstractUpdateCoordinator<Integer, Point> {
 
-    private final IPointStore<Q> store;
+    private final IPointStore<Point> store;
 
-    public PointStoreCoordinator(IPointStore<Q> store) {
+    public PointStoreCoordinator(IPointStore<Point> store) {
         checkNotNull(store, "store must not be null");
         this.store = store;
     }
 
     @Override
     public Integer initUpdate(double[] point, long sequenceNumber) {
-        return store.add(point, sequenceNumber);
+        int result = store.add(point, sequenceNumber);
+        return (result >= 0) ? result : null;
     }
 
     @Override
     public void completeUpdate(List<UpdateResult<Integer>> updateResults, Integer updateInput) {
-        updateResults.forEach(result -> {
-            result.getAddedPoint().ifPresent(store::incrementRefCount);
-            result.getDeletedPoint().ifPresent(store::decrementRefCount);
-        });
-        store.decrementRefCount(updateInput);
-        totalUpdates++;
+        if (updateInput != null) {
```

Comment:
Now that we've added a method `getStore` with a null implementation, and we're add support for do-nothing implementations of `initUpdate` and `completeUpdate`, it seems like we need to rethink the entire update-coordinator paradigm. If it's not a universal paradigm that we can use to define different update methods, then it's just adding noise to the library.

Reply:
The null check here is enable (i) count the updates only when they are completed (ii) count the updates for all the points in case of the internal shingling -- so that we can check that they are in sequential increasing order, increasing by 1. We can remove the null check if we relax either (i) and (ii). The real issue is that for internal shingling we will not generate a point (hence any non-null point to update) while the initial shingle is being filled.

Reply:
removing this check since now update executor checks for null for shingling issue